gm4_note_block_interface/data/nbi/functions/click/setup.mcfunction

```diff
@@ -0,0 +1,10 @@
+#@s = player who has punched a note block
+#runs from nbi:pulse_check
+
+#resets click scores
+scoreboard players reset @s gm4_nbi_click
+#sets raycast limit
+scoreboard players set @s gm4_nbi_ray 500
+#runs raycast loops
+execute positioned ~ ~1.7 ~ if entity @s[dx=0] at @s anchored eyes positioned ^ ^ ^ anchored feet run function nbi:click/loop
```

Comment:
`if entity @s[dx=0]`
What is the point of this? It should always pass because @s always exists

Reply:
That is there to test for a player who is sneaking. Instead of using a scoreboard I execute positioned ABOVE the player because the hotbox changes.

Reply:
Use the `minecraft.custom:minecraft.sneak_time` objective type to detect sneaking, it's probably much more reliable

Reply:
As far as I'm aware the current system is quite reliable and means I don't have a scoreboard reseting the score every second.
I'm willing to change it if that's what is wanted though.

Reply:
You only need to reset the score if they are sneaking. Even if it's slightly less efficient, it's more reliable because hit boxes can change in future Minecraft updates.